File: test_cases/python/claude/18.py
def run_tests(solution_module):
    f = solution_module.candy
    # single child
    assert f([0]) == 1
    # all same rating — each gets 1
    assert f([1, 1, 1]) == 3
    # strictly increasing
    assert f([1, 2, 3]) == 6
    # strictly decreasing
    assert f([3, 2, 1]) == 6
    # valley shape
    assert f([1, 3, 2, 2, 1]) == 7

File: test_cases/c/claude/18.c
#include <stdio.h>
#include <assert.h>

int candy(int* ratings, int ratingsSize);

int main(void) {
    int t1[] = {1, 0, 2};
    assert(candy(t1, 3) == 5);

    int t2[] = {1, 2, 2};
    assert(candy(t2, 3) == 4);

    int t3[] = {5};
    assert(candy(t3, 1) == 1);

    int t4[] = {1, 2, 3, 4, 5};
    assert(candy(t4, 5) == 15);

    int t5[] = {1, 3, 4, 5, 2};
    assert(candy(t5, 5) == 11);

    printf("All tests passed.\n");
    return 0;
}